src/server/scripts/Northrend/zone_sholazar_basin.cpp

```diff
@@ -772,6 +772,162 @@ class spell_shango_tracks : public SpellScriptLoader
+                    case EVENT_DROSTAN_SAY_7:
+                        Talk(DROSTAN_SAY_7, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_6:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_2, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_11:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_3, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_21:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_4, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_28:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_5, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_SLEEP, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_35:
+                        me->DespawnOrUnsummon(10s);
+                        Talk(DROSTAN_SAY_6, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_SAY_7, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_41:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_8, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_49:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_9, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                    case EVENT_DROSTAN_OBJECTIVE_56:
+                        me->DespawnOrUnsummon(4s);
+                        Talk(DROSTAN_SAY_10, player);
+                        _events.ScheduleEvent(EVENT_DROSTAN_RUN_AWAY, 3s);
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap _events;
+        ObjectGuid _playerGUID;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_drostanAI(creature);
+    }
+};
+
+class q12592_objective_tracker : public PlayerScript
+{
+public:
+    q12592_objective_tracker() : PlayerScript("") { }
+
+    void OnQuestObjectiveProgress(Player* player, Quest const* quest, uint32 /*objectiveIndex*/, uint16 progress) override
+    {
+        if (quest->GetQuestId() != QUEST_THE_GREAT_HUNTERS_CHALLENGE)
+            return;
+
+        switch (progress)
```

Comment:
how is the progress incremented ? like is it the number of creatures killed or items picked up or ... ?

Reply:
The quest requires to kill 60 creatures, so that's what the progress variable represents in this case.